### PR TITLE
Introduce BlockRecordProtocol as a subset of BlockRecord that the mempool manager uses for peak

### PR DESCRIPTION
**Pull request by AmineKhaldi**

### CODE REVIEW

**Comment by arvidn** [comment]
I think this looks like an improvement

File: chia/clvm/spend_sim.py [hunk 1]
```diff
@@ -187,8 +186,7 @@ async def close(self) -> None:
         await self.db_wrapper.close()
 
     async def new_peak(self) -> None:
-        # Create a protocol to make BlockRecord and SimBlockRecord interchangeable.
-        await self.mempool_manager.new_peak(self.block_records[-1], None)  # type: ignore[arg-type]
```
Comment (arvidn) [inline chia/clvm/spend_sim.py:191]: you could simplify `benchmarks/mempool.py` as well

Comment (AmineKhaldi) [reply chia/clvm/spend_sim.py:191]: Done.

**Review by arvidn — COMMENTED** [review]
this looks good. Would you mind if we hold off landing this until https://github.com/Chia-Network/chia-blockchain/pull/14635 lands? there will be some conflicts

**Comment by AmineKhaldi** [comment]
> this looks good. Would you mind if we hold off landing this until #14635 lands? there will be some conflicts

Yep, no worries.

**Review by arvidn — APPROVED** [review]

**Pull request merged**